src/main/java/com/grafana/opentelemetry/OpenTelemetryConfig.java

```diff
@@ -45,13 +45,34 @@ public MeterRegistry openTelemetryMeterRegistry(OpenTelemetry openTelemetry, Clo
 
     @Bean
     public OpenTelemetry openTelemetry(Optional<AutoConfiguredOpenTelemetrySdk> sdk) {
-        return sdk.<OpenTelemetry>map(AutoConfiguredOpenTelemetrySdk::getOpenTelemetrySdk)
-                       .orElse(OpenTelemetry.noop());
+        OpenTelemetry openTelemetry = sdk.<OpenTelemetry>map(AutoConfiguredOpenTelemetrySdk::getOpenTelemetrySdk)
+                .orElse(OpenTelemetry.noop());
+        addLogAppender(openTelemetry);
+        return openTelemetry;
+    }
+
+    private void addLogAppender(OpenTelemetry openTelemetry) {
+        //the openTelemetry object is not used yet, but it will be used in the future, when the global otel instance is not used by default anymore
+
+        try {
+            Class.forName("ch.qos.logback.classic.Logger");
+            LogbackConfig.addLogbackAppender();
+        } catch (ClassNotFoundException e) {
+            //not using logback
+            try {
+                Class.forName("org.apache.logging.log4j.core.LoggerContext");
+                Log4jConfig.addLog4jAppender();
+            } catch (ClassNotFoundException ex) {
+                // not using log4j2
+            }
+
+            logger.warn("no logging library found - OpenTelemetryAppender not added");
```

Comment:
If you don't change the structure of the methods (see my above comment) then I think this warn statement should be within the second catch block.  In its current location, it will run regardless of whether `forName()` throws an exception for `org.apache.logging.log4j.core.LoggerContext`.